### DISCUSSION TITLE
Idea: "dagger <action>" syntactic sugar

### DISCUSSION THREAD

**Discussion by aluzzardi** [dagger/dagger#2425] [Ideas]
Inspired by `yarn` and `npm`.

What if, in addition to `dagger do <action>`, we could also type `dagger <action>` for actions that don't clash with built-in command names?

```
dagger do deploy --prod
dagger deploy --prod
```

The inspiration comes from yarn/npm which do the same:

```
yarn run build
yarn build  ## build is actually not a yarn command
```

This would take one step further the "action inputs as flags" initiative: the entire command becomes user configurable.

The downside is clash with built-ins: if tomorrow we add a `dagger deploy` command, suddenly the user will have to type `dagger do deploy` instead.

While it is a major problem:

- There's always the `dagger do <action>` escape hatch, it's a recoverable error
- We already have this problem with flags, we need to mitigate it already anyway (and by the way, we should have an equivalent escape hatch here as well, e.g. `--set`)
- Mitigation could be using the "long" syntax from CI/automation (for instance, the GitHub action would call `dagger do`)
- Mitigation also includes versioning: we would not add new commands on patch versions. CI/automation is pinned to a minor version already.

/cc @shykes @helderco @jlongtine

**Comment by shykes** [comment]
I can see the appeal, but this feature is like a one-way door: once we cross it, we can never go back. At this early stage I don't have enough confidence that we wouldn't regret it...

I suggest that we leave this on the shelf and revisit in a few months.

**Comment by jlongtine** [comment]
I like this idea, but given the difficulty reversing this decision, it seems reasonable to tap the brake and revisit in a bit.
